src/index.js

```diff
@@ -134,9 +134,7 @@ function makeQuery(options) {
         // Cancel any side-effects
         query.cancelled = cancelledError
 
-        if (query.promise && query.promise.cancel) {
```

Comment:
`query.promise.cancel` would never get called because `query.promise` is declared within `react-query` as a promise that signifies the completion of the entire query. It is not the promise returned by the user provided `queryFn`.